enarx-keep-sgx/src/main.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![deny(clippy::all)]
+
+mod builder;
+mod component;
+mod convert;
+mod paged;
+
+use addr::{Address, Offset};
+use enarx_keep::{Event, Keep, Start};
+use span::Span;
+use units::bytes;
+
+use std::io::Result;
+
+extern "C" {
+    #[no_mangle]
+    fn enarx_eenter(tcs: Address<usize>);
```

Comment:
Ah, so this is the `%rdi` from `enter.s`.

Reply:
Yes, the first argument on 64-bit linux is `%rdi`. See here for more details: https://en.wikipedia.org/wiki/X86_calling_conventions#System_V_AMD64_ABI